shell/components/form/LabeledSelect.vue

```diff
@@ -50,10 +50,6 @@ export default {
       default: 'label',
       type:    String
     },
-    options: {
-      default:   null,
-      type:      Array
-    },
```

Comment:
I'm opting to define the `options` & `searchable` props in `shell/mixins/labeled-form-element.ts` for both `LabeledSelect.vue` and `Select.vue`. Doing so solves two problems:

1. Working with typescript is infinitely less complex. We don't have to do anything fancy in our mixin to state that the vue instance is sometimes a `LabeledSelect` or sometimes a `LabeledInput` and may or may not contain said properties..
2. The relationship between mixin and components that consume said mixins much more clear. It's not really a great design for mixins to have knowledge of the components that consume them 

Overall, I suggest that we consider a refactor of `shell/mixins/labeled-form-element.ts` as it might have too many responsibilities.

Reply:
1. Optional values should be omitted and stick to exclusively intended values, otherwise the code gets bloated. The `select` has no way to be searched, so it should not be there, as well as all the logic related to that.
2. It would be way more convenient to have a mixin for all the inputs as a kind of common API, then extend only some cases like `options` and `searchable` like here to their related components. I am not so sure how this gets translated inside Storybook and API generated though.

About Typescript, we will have definitely to rewrite all the props later on in any case, e.g. this array does not specify the type and you will not get any autocomplete or correction. Thankfully this would be possible after we migrate to a proper and modern Typescript setup.